src/identity/quartzOrganizations/actions/thunks/index.ts

```diff
@@ -13,73 +16,95 @@ import {
+  DefaultOrgStateError = 'DefaultOrgStateError',
+  DefaultOrgNetworkError = 'DefaultOrgNetworkError',
+}
+
+export class DefaultOrgStateError extends Error {
   constructor(message) {
     super(message)
-    this.name = 'DefaultOrgNotFoundError'
+    this.name = ThunkErrorNames.DefaultOrgStateError
   }
 }
 
-// Error Reporting
-import {reportErrorThroughHoneyBadger} from 'src/shared/utils/errors'
+export class DefaultOrgNetworkError extends Error {
+  constructor(message) {
+    super(message)
+    this.name = ThunkErrorNames.DefaultOrgNetworkError
+  }
+}
 
-export const getQuartzOrganizationsThunk = () => async (
+// Thunks
+export const getQuartzOrganizationsThunk = (accountId: number) => async (
   dispatch: Dispatch<Actions>,
   getState: GetState
 ) => {
   try {
     dispatch(setQuartzOrganizationsStatus(RemoteDataState.Loading))
-    const quartzOrganizations = await fetchQuartzOrgs()
+
+    const quartzOrganizations = await fetchOrgsByAccountID(accountId)
 
     dispatch(setQuartzOrganizations(quartzOrganizations))
   } catch (err) {
-    dispatch(setQuartzOrganizationsStatus(RemoteDataState.Error))
-
     reportErrorThroughHoneyBadger(err, {
       name: 'Failed to fetch /quartz/orgs/',
       context: {state: getState()},
     })
   }
 }
 
-export const updateDefaultOrgThunk = (newDefaultOrg: DefaultOrg) => async (
+export const updateDefaultOrgThunk = ({
+  accountId,
+  newDefaultOrg,
+}: UpdateOrgParams): AppThunk<Promise<void>> => async (
   dispatch: Dispatch<Actions>,
   getState: GetState
-) => {
+): Promise<void> => {
   try {
     dispatch(setQuartzOrganizationsStatus(RemoteDataState.Loading))
 
-    await updateDefaultQuartzOrg(newDefaultOrg.id)
+    await updateDefaultOrgByAccountID({
+      accountNum: accountId,
+      orgId: newDefaultOrg.id,
+    })
 
     dispatch(setQuartzDefaultOrg(newDefaultOrg.id))
 
     const state = getState()
     const orgStatus = state.identity.currentIdentity.status
 
     if (orgStatus === RemoteDataState.Error) {
-      const defaultOrgErrMsg =
-        'quartzOrganizations state does not contain requested default organization'
-
-      reportErrorThroughHoneyBadger(new OrgNotFoundError(defaultOrgErrMsg), {
-        name: defaultOrgErrMsg,
-        context: {
-          org: newDefaultOrg,
-          state: getState(),
-        },
-      })
+      throw new DefaultOrgStateError(ThunkErrorNames.DefaultOrgStateError)
     }
   } catch (err) {
-    dispatch(setQuartzOrganizationsStatus(RemoteDataState.Error))
-
     reportErrorThroughHoneyBadger(err, {
-      name: 'Failed to update /quartz/orgs/default',
-      context: {state: getState()},
+      name: err.name,
+      context: {
+        message: err.message,
+        org: newDefaultOrg,
+        state: getState(),
+      },
     })
+
+    switch (err.name) {
+      case ThunkErrorNames.DefaultOrgStateError:
+        throw new DefaultOrgStateError(err)
+      default:
+        throw new DefaultOrgNetworkError(err)
+    }
```

Comment:
Will the UI properly catch and handle these errors?

Reply:
This could be a YAGNI situation since the UI doesn't care which error is thrown (we only need this for error reporting). I think for now, we can just throw a generic `Error` to the UI, which is all that's needed.